### DISCUSSION TITLE
Why is this education verification process so bad?

### DISCUSSION THREAD

**Discussion by linpatr** [community/community#134800] [GitHub Education]
### Hi everyone,

GitHub's student verification process for its Education program is fundamentally broken. It appears designed to obstruct rather than facilitate student access. The issues are numerous and severe:

Restrictive Document Requirements:
The system demands specific documents that, for some reason, cannot be uploaded directly through a file upload.

Misguided Location Verification:
GitHub erroneously assumes all legitimate students must be near a physical campus, completely disregarding the reality of e-campus and distance learning. How the hell are we supposed to definitively prove we are e-campus students?

Arbitrary Camera Restrictions:
The insistence on using only a front-facing camera for verification is both illogical and unnecessarily limiting.

Inefficient and Limiting Request System:
The process is so broken that it often requires multiple attempts, yet GitHub imposes a strict limit on verification requests. I've been told I've "already submitted 4 requests" when a functional system should have required only one attempt.

These policies do not serve any legitimate verification purpose. Instead, they create needless barriers.

GitHub needs to overhaul this system immediately. A truly effective verification process should be inclusive, flexible, and mindful of diverse student circumstances. The current system is none of these things.

The verification process is not just flawed, it's actively preventing verification. Stop hindering students who are trying to learn and contribute to the development community.

Completely and utterly disgraceful.

**Comment by ZaHumph** [comment]
I am having the same issues, have verified my information as well as myself multiple times but I'm still getting the reverification issue which is stupid

**Comment by Valeriucoding** [comment]
I agree, they've complicated the whole process. I applied so easily two years ago and now I'm yet to get verified, I'm trying my luck with the support team. The whole process is a mess and the verification form on the mobile is broken.

**Comment by ohubenko** [comment]
I agree with that, previously concept of verification was totally clear. Now I can't be verified. Verification from phone is broken, only front camera is available.

**Comment by ZaHumph** [comment]
I was verified before but the process is horrible

**Comment by jthrilly** [comment]
Agree with everything that has been said. And to top it off, they have removed the ability to create support requests to deal with these issues. Utterly, utterly absurd.

In my case I was flagged as being not on campus, which is correct (I am remote, and based in an entirely different country). How exactly am I supposed to "prove" the reason for that using _my webcam_?

**Reply by Valeriucoding** [reply]
I get rejected on the same stupid reason. I provided a document that shows that I'm enrolled, but I don't have one that proves I'm a distance learner. Talking with support feels like going in a circle.

**Comment by ZaHumph** [comment]
I'm going to try again not sure what they are requesting

**Comment by yanuu1337** [comment]
I am running into a similiar issue. My country has paper double-sided school IDs that are filled manually using a pen and stamps. I have given translations as required, but the system that handles verification doesn't seem to recognize that. 
It's stupid that manual verification was stopped.
While I might understand that a robot may not accept a paper ID, what I don't understand is why you can't upload screenshots as proof - in Poland you can request an online ID available on your phone, but apparently Github thinks it's not valid lol.
I am also deeply disappointed by the FAQ; It doesn't say anything on double-sided IDs besides "You figure out the logistics for that", yet you can't upload two images.

**Comment by Tsteib** [comment]
I am happy to learn it's not just me. I am trying to verify myself as an educator and I keep getting rejected.

**Comment by Daar543** [comment]
I wonder if the photo is actually saved on the server side during the shoot, or you can technically manipulate the DOM to send any image in the "take photo". Apparently drag and drop could work.

**Comment by mindaika** [comment]
Not just bad: stunningly bad. I was asked to prove my status as a student, not by using best practices, like ID.me which works everywhere else, but by using the front camera only on my phone to take a picture of a PDF on my computer. I was then asked to "prove why I wasn't on [the remote] campus." There is of course no way to contact Github support.

I struggle to understand how anyone could have UATd this process and found it to be acceptable. On the plus side, I was trying to sign up for the Dev Pack so I could try out CoPilot before I paid for it. After this experience, I would never pay for it, so question answered

**Reply by sfrezza** [reply]
And... Even when you get their required .jpg of the documents, they fail you for it without useful comments. Just "Please use your device camera to submit academic affiliation documents." 
Exceptionally bad use case implementation.

**Comment by stickyburn** [comment]
Its easily the worst experience I've ever had.. How do they expect us to take pictures of transcripts using a laptop webcam? I've also never seen a school ID that has enrollment dates written on it like have they ever been to school?!

On top of this it somehow thinks im not on campus.., sorry Github my bad for living 1km away in an apartment, but then it wants picture as proof? Like wtf am I supposed to show you?? My kitchen or me eating a goddamn burger?
Its obvious they dont care about students, if they did a simple email verification would've sufficed. Too bad 99% of the developers are never opting for the pro features, I know I certainly will not after this godforsaken experience.

**Comment by averwhy** [comment]
Completely and utterly disgraceful from this company. Fix this PLEASE! Even using OBS virtual camera doesn't work because as soon as you take the photo it just goes completely blurry. 

Just absolutely mind boggling that this made it through QA to production. This is 100% the worst application process of any kind, shape or form that I have ever had the displeasure of experiencing

**Comment by Valeriucoding** [comment]
From some other threads I've seen over here, some people tried altering the document that they were showing to say that they are learning remotely and they got through. Do this at your own risk tho, I don't know how legal it is.

**Reply by averwhy** [reply]
Remember kids, if it's Microsoft, its Moral

**Reply by Valeriucoding** [reply]
Don't forget Adobe!

**Comment by YifeiZhang123** [comment]
Though i got the approved,but i cant get the benefits.Its so bad

**Reply by Valeriucoding** [reply]
You'll have to wait for max two weeks after you get approved to get your benefits.

**Comment by Shifath472533** [comment]
Absolutely dumb and rubbish process.

**Comment by totallyeli** [comment]
I used my University VPN and then I was able to upload my enrollment letter. After that it got approved.
<img width="923" height="45" alt="firefox_jM9rswHp2w" src="https://github.com/user-attachments/assets/ace6162e-1271-4875-85db-b8452a0181e9" />

**Comment by dane-anderson** [comment]
What an awful process.  Also, have tired to get approved 5 times.   I meet all the requirements and have uploaded on the camera!  Who uses a camera to send paper documents?   also, zero help on the website.

**Comment by tilman-schieber** [comment]
It is really frustrating.

On desktop, I can upload a document, but it never seems to register. After the upload finishes, the document simply disappears.

Why does the system only accept images? I already have a digitally signed PDF proving my affiliation, but I have to take a screenshot of it just to upload it. Even then, as mentioned above, the uploaded file just disappears.

On mobile, the app opens the front-facing camera with a countdown timer, as if I'm supposed to take a selfie. Even after switching to the rear camera on my iPhone, the photos are captured at such a low resolution that the document is completely unreadable. As a result, my verification has now been rejected for the sixth time.

However, even if Microsoft claims you are not able to create a support ticket for this horrible process anymore. It's possible by selecting topic: other. It also got answered. While the answer was only unhelpful boilerplate advice I was able to reply back and let's hope this leads to something.

**Reply by akes44** [reply]
Is there any one from GitHub addressing this?

**Reply by sfrezza** [reply]
Apparently not. These were the same issues I had to deal with last year.

**Reply by tilman-schieber** [reply]
After three messages with their very friendly customer support, they finally accepted my application. If you’re having the same problem, I can only recommend creating a support ticket. If they see how much support this flawed process generates, it might spur them into action.
